io/src/main/scala/fs2/io/package.scala

```diff
@@ -0,0 +1,104 @@
+package fs2
+
+import fs2.util.{Async, Effect}
+import fs2.util.syntax._
+import java.io.{InputStream, OutputStream}
+
+package object io {
+
+  private[io] def readBytesFromInputStream[F[_]](is: InputStream, buf: Array[Byte])(implicit F: Effect[F]): F[Option[Chunk[Byte]]] =
+    F.delay(is.read(buf)).map { numBytes =>
+      if (numBytes < 0) None
+      else if (numBytes == 0) Some(Chunk.empty)
+      else Some(Chunk.bytes(buf, 0, numBytes))
+    }
+
+  private[io] def readInputStreamGeneric[F[_]](fis: F[InputStream], chunkSize: Int, f: (InputStream, Array[Byte]) => F[Option[Chunk[Byte]]], closeAfterUse: Boolean = true)(implicit F: Effect[F]): Stream[F, Byte] = {
+    val buf = new Array[Byte](chunkSize)
+
+    def useIs(is: InputStream) =
+      Stream.eval(f(is, buf))
+        .repeat
+        .through(pipe.unNoneTerminate)
+        .flatMap(Stream.chunk)
+
+    if (closeAfterUse)
+      Stream.bracket(fis)(useIs, is => F.delay(is.close()))
+    else
+      Stream.eval(fis).flatMap(useIs)
+  }
+
+  /**
+    * Reads all bytes from the specified `InputStream` with a buffer size of `chunkSize`.
+    * Set `closeAfterUse` to false if the `InputStream` should not be closed after use.
+    *
+    * Blocks the current thread.
+    */
+  def readInputStream[F[_]](fis: F[InputStream], chunkSize: Int, closeAfterUse: Boolean = true)(implicit F: Effect[F]): Stream[F, Byte] =
+    readInputStreamGeneric(fis, chunkSize, readBytesFromInputStream[F], closeAfterUse)
+
+  /**
+    * Reads all bytes from the specified `InputStream` with a buffer size of `chunkSize`.
+    * Set `closeAfterUse` to false if the `InputStream` should not be closed after use.
+    *
+    * This will block a thread in the current `Async` instance, so the size of any associated
```

Comment:
Also I was thinking, are there really scenarios where you want the IS to outlive the resulting process? I mean is this absolutelly necessary? I am just wondering if this can't be source of just hard to track bugs and resource leaks.